Index sbt plugins from Bintray repository

adpi2 opened the pull request:
Fixes #620 

It enables pulling packages that are linked to the [sbt/sbt-plugin-releases](https://bintray.com/sbt/sbt-plugin-releases) repository. About 300 packages are currently linked packages and so they have been ignored so far.

It is ready for review but it will require member rights from the Bintray sbt organization to be run effectively because the rate-limit for non-member is too low (1400 requests per month).

Commit descriptions:
- [6e6ec63 manage Bintray client](https://github.com/scalacenter/scaladex/commit/6e6ec6399d0ca9681dea5dc494c5f9a798141516): Refactoring: Use scala-arm to manage Play `AhsWSClient`s and `BintrayClient`s.
- [d47c5b3 update lnked sbt-plugins from Bintray](https://github.com/scalacenter/scaladex/commit/d47c5b30d947a177a3eb92fb62515e017bec06c1): Rewrite the `BintrayDownloadSbtPlugins` class as `UpdateBintraySbtPlugins` to group packages by Bintray repository (linked packages belong to another repository) and then download ivy files of packages that have been modified since last update.

julienrf commented on data/src/main/scala/ch.epfl.scala.index.data/download/PlayWsDownloader.scala:
```diff
@@ -57,10 +34,9 @@ trait PlayWsDownloader {
-      client => f(client).andThen { case _ => client.close() }
-    )
+  def managed[A](f: WSClient => Future[A]): Future[A] = {
+    PlayWsClient.open().toFuture.flatMap(f)
```
Where is the `toFuture` method defined? i’m looking at http://jsuereth.com/scala-arm/latest/api/index.html#resource.ManagedResource but that may not be the right place…?

adpi2 replied:
It should be there... but it's not, so I think the scaladoc is not up-to-date.
You can find the `toFuture` documentation [here](https://github.com/jsuereth/scala-arm/blob/7a157f96cf2a906d803c6a7110e68145a44c61ca/src/main/scala/resource/ManagedResource.scala#L112-L118).

julienrf commented on data/src/main/scala/ch.epfl.scala.index.data/bintray/SbtPluginsData.scala:
```diff
@@ -59,3 +50,46 @@ case class SbtPluginReleaseModel(
+          dep.getOrganisation,
+          dep.getName,
+          dep.getRevision,
+          scope = /* TODO */ None,
```
Is this a change with the previous behavior?

adpi2 replied:
No it does not. It is code that I have moved to this place for readability.

julienrf commented on data/src/main/scala/ch.epfl.scala.index.data/bintray/UpdateBintraySbtPlugins.scala:
```diff
@@ -0,0 +1,165 @@
+                              scalaVersion,
+                              sbtVersion,
+                              version) =>
+        Future {
```
Out of curiosity: why do we want to do that in a separate execution thread?

adpi2 replied:
Mostly because the `bintray.downloadUrl(...)` at line 117 is doing IO that we want to parallelize (for each new release).

adpi2 commented:
I have added comments in the important places and around the tricky code of this PR as well as some detailed commit descriptions.

The most important part is the `UpdateBintraySbtPlugins` class which has been written from scratch and replace the removed `BintrayDownloadSbtPlugins` class. It changes the previous behavior to handle the packages that are not published to the Bintray repo but linked to it.

Also I have added a third commit to update the github infos of the sbt-plugins. If a new release is found, the Github info of the corresponding project, based on the Bintray vcs_url field, are updated. This allows to skip the `github` step when updating the sbt plugins. The `github` step is not viable anymore because it tries to update all and every project of Scaladex which will eventually hit the Github rate limit.
